### PR TITLE
new resource quota rules

### LINKED ISSUE #136: [New Resource]: /storage/quota/rules
### Description

/storage/quota/rules

### Requested Resource(s) and/or Data Source(s)

netapp-ontap_storage_quota_rules

### Potential Terraform Configuration

_No response_

### References

https://docs.netapp.com/us-en/ontap-restapi/ontap/swagger-ui/index.html#/storage/quota_rule_create

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by suhasbshekar**
#136

### CODE REVIEW

File: internal/provider/storage/storage_quota_rules_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	ntest "github.com/netapp/terraform-provider-netapp-ontap/internal/provider"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccStorageQuotaRulesResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { ntest.TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: ntest.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
```
Comment (chuyich) [inline internal/provider/storage/storage_quota_rules_resource_test.go:17]: Can we have a test with the required parameters only if it makes sense?

Comment (suhasbshekar) [reply internal/provider/storage/storage_quota_rules_resource_test.go:17]: the current test covers all scenarios, so it should be fine I think

File: examples/resources/netapp-ontap_storage_quota_rules/resource.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    name = "lunTest"
+    }
+  svm = {
+    name = "carchi-test"
+    }
+  type = "tree"
+  qtree = {
+    name = ""
+    }
+  # users = [{
```
Comment (wenjun666) [inline examples/resources/netapp-ontap_storage_quota_rules/resource.tf]: can you either remove the commented code or add explanation why it is commented.

File: internal/provider/storage/storage_quota_rules_resource_test.go (hunk 1 above)
Comment (chuyich) [reply internal/provider/storage/storage_quota_rules_resource_test.go:17]: ok. I just want to make sure if the user provides only required, very limited basic parameters and it should work.

File: internal/provider/storage/storage_quota_rules_resource.go [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+		"hard_limit": types.Int64Value(restInfo.Files.HardLimit),
+		"soft_limit": types.Int64Value(restInfo.Files.SoftLimit),
+	}
+	objectValue, diags := types.ObjectValue(elementTypes, elements)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+	}
+	data.Files = objectValue
+	data.ID = types.StringValue(restInfo.UUID)
+
```
Comment (wenjun666) [inline internal/provider/storage/storage_quota_rules_resource.go:255]: Is 'user' missing?

Comment (suhasbshekar) [reply internal/provider/storage/storage_quota_rules_resource.go:255]: Nope, not required here, I tested it again, its good, successfully creating, modifying, deleting and updating the state file

File: examples/resources/netapp-ontap_storage_quota_rules/resource.tf (hunk 2 above)
Comment (suhasbshekar) [reply examples/resources/netapp-ontap_storage_quota_rules/resource.tf]: removed it

File: internal/provider/storage/storage_quota_rules_resource.go (hunk 3 above)
Comment (carchi8py) [reply internal/provider/storage/storage_quota_rules_resource.go:255]: Is user not returned then?

Comment (suhasbshekar) [reply internal/provider/storage/storage_quota_rules_resource.go:255]: user can be added only through input and when used as input state is updated, if not, its always empty and its not modifiable too

**Review by chuyich — APPROVED** [review]

**Pull request merged**